streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -213,6 +212,32 @@ public <K, V> void send(final String topic,
                     keyClass,
                     valueClass),
                 exception);
+        } catch (final SerializationException exception) {
+            final ProducerRecord<K, V> record = new ProducerRecord<>(topic, partition, timestamp, key, value, headers);
+            final ProductionExceptionHandler.ProductionExceptionHandlerResponse response;
+
+            log.debug("Error serializing record to topic {} due to {}", topic, exception);
+
+            try {
+                response = productionExceptionHandler.handleSerializationException(record, exception);
+            } catch (final Exception e) {
+                log.error("Fatal when handling serialization exception", e);
+                recordSendError(topic, e, null);
+                return;
+            }
+
+            if (response == ProductionExceptionHandlerResponse.FAIL) {
+                log.error("Fatal due to SerializationException.", exception);
+                droppedRecordsSensor.record();
+                throw exception;
+            }
+
+            log.warn("Unable to serialize record, continue processing. " +
+                    "ProducerRecord(topic=[{}], partition=[{}], timestamp=[{}])",
+                topic,
+                partition,
+                timestamp);
+
```

Comment:
Just realizing that a `return` is missing here -- without `return` we would still call `streamsProducer.send` passing in `key == null` and `value == null` but we should not send anything...

I can push a fix into this PR (including extending the tests for it) and merge afterwards.

Reply:
oh god, thanks for catching this.

Reply:
might be worth adding a test there actually.